packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -1129,9 +1129,12 @@ class Padding extends SingleChildRenderObjectWidget {
 
   @override
   void updateRenderObject(BuildContext context, RenderPadding renderObject) {
+    final TextDirection textDirection = Directionality.of(context);
+    if (textDirection != null)
+      renderObject.textDirection = textDirection; // in case it was null before, we must set it before anything that depends on it
```

Comment:
Hum...  This pattern is pretty ugly.  Here are two alternatives:

1. Delay resolving the padding/alignment until the resolve value is actually needed.  This will give both writes a chance to land before we resolve (avoiding the problem of resolving in an inconsistent state).

2. Add a method to set the padding/alignment and textDirection at the same time.  That will like the client perform an atomic write of both properties an avoid the inconsistent state entirely.

Reply:
Number 1 is my preferred approach, but it means being more aggressive about marking things dirty. I'll try that.